rbac/management/workspace/utils/access.py

```diff
@@ -0,0 +1,272 @@
+        target_workspace: The workspace ID to check, or None for list operations
+
+    Returns:
+        bool: True if the user has permission, False otherwise
+    """
+    # Get org_id from request for feature flag targeting
+    org_id = getattr(request.user, "org_id", None)
+    if FEATURE_FLAGS.is_workspace_access_check_v2_enabled(org_id=org_id):
+        return is_user_allowed_v2(request, required_operation, target_workspace)
+    return is_user_allowed_v1(request, required_operation, target_workspace)
+
+
+def is_user_allowed_v1(request, required_operation, target_workspace):
+    """
+    Check if the user is allowed using V1 workspace access logic.
+
+    This is the legacy implementation using direct role/permission checks.
+    Use WorkspaceAccessPermission for the main entry point which handles
+    V1/V2 feature flag branching.
+
+    Args:
+        request: The HTTP request object
+        required_operation: The operation to check ("read" or "write")
+        target_workspace: The workspace ID to check, or None for list operations
+
+    Returns:
+        bool: True if the user has permission, False otherwise
+    """
+    root_workspace_id = str(Workspace.objects.root(tenant_id=request.tenant).id)
+    is_get_action = request.method == "GET"
+    if target_workspace is None:
+        # If the target workspace is not provided, check if the user has the required permission
+        # on any workspace.
+        target_workspace = root_workspace_id
+    if required_operation == "read":
+        allowed_operations = ["read", "write", "*"]
+    else:
+        allowed_operations = ["write", "*"]
+    valid_perm_tuples = set()
+    for valid_resource in ["groups", "*"]:
+        for valid_operation in allowed_operations:
+            valid_perm_tuples.add((f"inventory:{valid_resource}:{valid_operation}", target_workspace))
+    tuple_set = workspace_permission_tuple_set(request, root_workspace_id, is_get_action)
+
+    if is_get_action:
+        # Get the set of permission tuples for later filter
+        request.permission_tuples = tuple_set
+    return any(valid_perm_tuple in tuple_set for valid_perm_tuple in valid_perm_tuples)
+
+
+def is_user_allowed_v2(request, required_operation, target_workspace):
+    """
+    Check if the user is allowed to perform the required permission on the target workspace using Inventory API.
+
+    This is the v2 implementation using Inventory API for permission checks.
+
+    Args:
+        request: The HTTP request object
+        required_operation: The operation/relation to check (view, create, edit, move, delete)
+        target_workspace: The workspace ID to check, or None for list operations
+
+    Returns:
+        bool: True if the user has permission, False otherwise
+    """
+    principal = get_principal_from_request(request)
+    # Format principal ID as required by Inventory API (e.g., "localhost/username")
+    principal_id = Principal.user_id_to_principal_resource_id(principal.user_id)
```

Comment:
Might need the task user_id to be unique to prevent any issues: https://issues.redhat.com/browse/RHCLOUD-38877